### ISSUE TITLE
nose test failing on read_bed_as_list.pyx

### ISSUE THREAD

**Issue by dputhier** [dputhier/pygtftk#98]
On 'make nose'.

    =================
    ./stats/intersect/read_bed/read_bed_as_list.py
    ......
    ----------------------------------------------------------------------
    Ran 6 tests in 0.460s

    OK
    =================
    ./stats/intersect/read_bed/read_bed_as_list.pyx
    F
    ======================================================================
    FAIL: Doctest: read_bed_as_list.pyx
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/Users/puthier/miniconda3/envs/pygtftk/lib/python3.6/doctest.py", line 2199, in runTest
        raise self.failureException(self.format_failure(new.getvalue()))
    AssertionError: Failed doctest test for read_bed_as_list.pyx
      File "/Users/puthier/miniconda3/envs/pygtftk/lib/python3.6/site-packages/pygtftk-1.0.5-py3.6-macosx-10.7-x86_64.egg/pygtftk/stats/intersect/read_bed/read_bed_as_list.pyx", line 0

    ----------------------------------------------------------------------
    File "/Users/puthier/miniconda3/envs/pygtftk/lib/python3.6/site-packages/pygtftk-1.0.5-py3.6-macosx-10.7-x86_64.egg/pygtftk/stats/intersect/read_bed/read_bed_as_list.pyx", line 56, in read_bed_as_list.pyx
    Failed example:
        npt.assert_array_equal(result[2], np.array(['chr1']))
    Expected:
        """
    Got nothing
    ----------------------------------------------------------------------
    File "/Users/puthier/miniconda3/envs/pygtftk/lib/python3.6/site-packages/pygtftk-1.0.5-py3.6-macosx-10.7-x86_64.egg/pygtftk/stats/intersect/read_bed/read_bed_as_list.pyx", line 127, in read_bed_as_list.pyx
    Failed example:
        assert result == OrderedDict([('chr1', 200), ('chr2', 300), ('all_chrom', 900)])
    Expected:
        """
        exclusion = exclusion.to_dataframe()
        for _, excl in exclusion.iterrows():
            excl_length = abs(excl['end'] - excl['start'])
            chromsizes[excl['chrom']] = chromsizes[excl['chrom']] - excl_length
        return chromsizes
    Got nothing
    ----------------------------------------------------------------------
    File "/Users/puthier/miniconda3/envs/pygtftk/lib/python3.6/site-packages/pygtftk-1.0.5-py3.6-macosx-10.7-x86_64.egg/pygtftk/stats/intersect/read_bed/read_bed_as_list.pyx", line 291, in read_bed_as_list.pyx
    Failed example:
        assert str(result) == 'chr1\t50\t80\nchr1\t9900\t10000\nchr2\t50\t100\nchr4\t100\t150\nchr5\t150\t250\nchr6\t1\t200\n'
    Expected:
        """
    Got nothing

**Comment by qferre** [comment]
I think the error is due to the strings in the nosetest. I'll fix it.

**Comment by qferre** [comment]
Nevermind, I just saw that you found it too and already fixed it :). I'll close this issue.

**Comment by dputhier** [comment]
I have already fixed some strings. But there may be some left to check.

**Comment by dputhier** [comment]
Check the test on your side. I have fixed tests but this one remains.

**Comment by qferre** [comment]
Okay that one is weird. The code for the unitary test passes when run manually in Ipython so at least we know it's not a code problem. I think it's a problem with the nose syntax. I'm working on it.

**Comment by qferre** [comment]
Fixed in e4153cdadc40b957de3f08b61b8e9a120836853d. It was the stupidest bug ever.

When writing a nose test, you must write this :

```
    >>> npt.assert_array_equal(result[2], np.array(['chr1']))
    
    """
```

Remembering to jump a line between the assert and the end of the string, otherwise it things that `"""` is the expected output of the `assert` for some retarded reason.

All nose tests are now passing for me. @dputhier : can you confirm they pass for you too ?

**Comment by dputhier** [comment]
OK :). Closing.
